src/app.py

```diff
@@ -336,7 +341,121 @@ def get_ancestor_organs(id):
+(dataset.status == 'Published'). For public collections only connected datasets that are
+public are returned with it.
+
+Parameters
+----------
+id : str
+    The HuBMAP ID (e.g. HBM123.ABCD.456) or UUID of target collection 
+
+Returns
+-------
+json
+    The collection detail with a list of connected datasets (only public datasets 
+    if user doesn't have the right access permission)
+"""
+@app.route('/collections/<id>', methods = ['GET'])
+def get_collection(id):
+    # Token is not required, but if an invalid token provided,
+    # we need to tell the client with a 401 error
+    validate_token_if_auth_header_exists(request)
+
+    # Use the internal token to query the target collection
+    # since public collections don't require user token
+    token = get_internal_token()
+
+    # Get the entity dict from cache if exists
+    # Otherwise query against uuid-api and neo4j to get the entity dict if the id exists
+    collection_dict = query_target_entity(id, token)
+
+    # A bit validation
+    if collection_dict['entity_type'] != 'Collection':
+        abort_bad_req("Target entity of the given id is not a collection")
+
+    # Try to get user token from Authorization header
+    # It's highly possible that there's no token provided
+    user_token = get_user_token(request)
+
+    # The user_token is flask.Response on error
+    # Without token, the user can only access public collections, modify the collection result
+    # by only returning public datasets attached to this collection
+    if isinstance(user_token, Response):
+        # When the requested collection is not public, send back 401
+        if ('registered_doi' not in collection_dict) or ('doi_url' not in collection_dict):
+            # Require a valid token in this case
+            abort_unauthorized("The requested collection is not public, a Globus token with the right access permission is required.")
+
+        # Otherwise only return the public datasets attached to this collection
+        # for Collection.datasets property
+        complete_dict = get_complete_public_collection_dict(collection_dict)
+    else:
+        # When the groups token is valid, but the user doesn't belong to HuBMAP-READ group
+        # Or the token is valid but doesn't contain group information (auth token or transfer token)
+        # Only return the public datasets attached to this Collection
+        if not user_in_globus_read_group(request):
+            complete_dict = get_complete_public_collection_dict(collection_dict)
+        else:
+            # We'll need to return all the properties including those
+            # generated by `on_read_trigger` to have a complete result
+            complete_dict = schema_manager.get_complete_entity_result(user_token, collection_dict)
+
+    # Will also filter the result based on schema
+    normalized_complete_dict = schema_manager.normalize_object_result_for_response( provenance_type='ENTITIES'
```

Comment:
Need to add `properties_to_include=['protocol_url']` to this method signature otherwise that property won't be grabbed from the Activity and returned with the rest of the entity properties. This is why a lot of the tests are failing:
`schema_manager.normalize_object_result_for_response('ENTITIES', complete_dict,
                                                                       properties_to_include=['protocol_url'])`